frontend/src/component/ai/AIChat.tsx

```diff
@@ -106,26 +104,34 @@ export const AIChat = () => {
         scrollToEnd();
     }, [open]);
 
-    const onSend = async (message: string) => {
-        if (!message.trim() || loading) return;
+    const onSend = async (content: string) => {
+        if (!content.trim() || loading) return;
 
         try {
             setLoading(true);
-            const tempMessages: ChatMessage[] = [
-                ...messages,
-                { role: 'user', content: message },
-                { role: 'assistant', content: '_Unleash AI is typing..._' },
-            ];
-            setMessages(tempMessages);
-            const newMessages = await chat(tempMessages.slice(0, -1));
+            setMessages((currentMessages) => [
+                ...currentMessages,
+                { role: 'user', content },
+            ]);
+            const { messages: newMessages } = await chat(content);
             mutate(() => true);
             setMessages(newMessages);
-            setLoading(false);
         } catch (error: unknown) {
+            setMessages((currentMessages) => [
+                ...currentMessages,
+                AI_ERROR_MESSAGE,
```

Comment:
Should we log this into the chat messages table as well? It could be weird since it's coming from Unleash and not the AI integration nor the user, but if we're presenting it to the user we might want to have it present in the log as well

Reply:
If Unleash is down that will be hard to do, is this handled server side as well?